src/OrchardCore.Modules/OrchardCore.Users/Controllers/AdminController.cs

```diff
@@ -224,7 +224,7 @@ public async Task<ActionResult> IndexPOST(UserIndexOptions options, IEnumerable<
             // Check a dummy user account to see if the current user has permission to manage it.
             var authUser = new User();
 
-            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageUsers, authUser))
+            if (!await CanEditContextAsync(authUser))
```

Comment:
At first I didn't understand what is `CanEditContextAsync` 

I guess keeping `_authorizationService.AuthorizeAsync` keeps code more in flow and readable rather than moving logic in private functions

Reply:
@ns8482e 
Here `CanEditContextAsync` is performing two check. First it checking to see if the user has EditUsers, or if the user has `ManageOwnUserInformation` to the given user.  changing it to `await _authorizationService.AuthorizeAsync(User, Permissions.EditUsers, user)` would be a different logic. We could rename the `CanEditContextAsync` method to `CanEditUserAsync` to make it more readable. Thoughts?